cumulusci/tasks/bulkdata/mapping_parser.py

```diff
@@ -78,18 +80,167 @@ def oid_as_pk_is_deprecated(cls, v):
         )
         return v
 
+    @validator("fields_")
+    def standardize_fields_to_dict(cls, values):
+        if values is None:
+            values = {}
+        if type(values) is list:
+            return {elem: elem for elem in values}
+
+        return values
+
+    @root_validator
+    def set_default_table(cls, values):
+        """Automatically populate the `table` key with `sf_object`, if not present."""
+        if values["table"] is None:
+            values["table"] = values.get("sf_object")
+
+        return values
+
     @root_validator  # not really a validator, more like a post-processor
     def fixup_lookup_names(cls, v):
         "Allow lookup objects to know the key they were attached to in the mapping file."
         for name, lookup in v["lookups"].items():
             lookup.name = name
         return v
 
+    def _is_injectable(self, element: str) -> bool:
+        return element.count("__") == 1
+
+    def validate_and_inject_namespace(
+        self,
+        org_config: OrgConfig,
+        namespace: str,
+        operation: DataOperationType,
+        inject_namespaces: bool = False,
+        drop_missing: bool = False,
+    ):
+        """Process the schema elements in this step.
+
+        First, we inject the namespace into object and field names where applicable.
+        Second, we validate that all fields are accessible by the running user with the
+        correct permission level for this operation.
+        Lastly, if drop_missing is True, we strip any fields that are not present (namespaced
+        or otherwise) from the target org.
+
+        Return True if this object should be processed. If drop_missing is True, a False return
+        value indicates we should skip this object. If drop_missing is False, a Fakse return
+        value indicates that one or more schema elements couldn't be validated."""
+
+        ret = True
+
+        def inject(element: str):
+            return f"{namespace}__{element}"
+
+        # Determine whether we need to inject our sObject.
+        if inject_namespaces and self._is_injectable(self.sf_object):
+            global_describe = {
+                entry["name"]: entry
+                for entry in org_config.salesforce_client.global_describe()["sobjects"]
```

Comment:
Funny that I can't find this method:

```
 $  cci org shell --python "print(org_config.salesforce_client.global_describe())"
2020-06-18 12:29:15: Getting scratch org info from Salesforce DX
Error: 'SFType' object is not callable
Run this command for more information about debugging errors: cci error --help
```

Reply:
That's because it is a mistake! Fixed that.